### PR TITLE
fix: clear function on non text inputs

### PR DESCRIPTION
**Pull request by Meemaw**
`element.setSelectionRange(0, element.value.length)` does not work on all input types (e.g. `type="number"` or `type"email"`). See https://stackoverflow.com/questions/33406169/failed-to-execute-setselectionrange-on-htmlinputelement-the-input-elements

Using `userEvent.clear` on such element will throw an error:
Screenshot 2020-05-08 at 15 16 08

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -106,29 +106,40 @@ function fireChangeEvent(event) {
...
+      inputType: "deleteContentBackward",
     });
     element.value = ""; // when we add special keys to API, will need to respect selected range
   }
 }
 
 function selectAll(element) {
   userEvent.dblClick(element); // simulate events (will not actually select)
+  const elementType = element.type;
+  if (isInputElement(element)) {
```
Comment (Meemaw) [inline src/index.js:132]: We have to check this because type is a read only property on textarea.

File: package.json [hunk 2]
```diff
@@ -33,25 +33,25 @@
...
+    "lint-staged": "^10.2.2",
+    "prettier": "^2.0.5",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "regenerator-runtime": "^0.13.5",
-    "semantic-release": "^17.0.4",
+    "semantic-release": "^17.0.7",
     "travis-deploy-once": "5.0.11",
     "vue": "^2.6.11",
     "vue-template-compiler": "^2.6.11"
```
Comment (kentcdodds) [inline package.json:57]: Was it necessary to update all dependencies for this PR?

Comment (Meemaw) [reply package.json:57]: No need, but I like to keep dependencies up to date. I can revert rhat if you'll like that.

Comment (Meemaw) [reply package.json:57]: ping @kentcdodds

**Comment by Meemaw** [comment]
@kentcdodds I've removed the `package.json` changes

File: src/index.js (hunk 1 above)
Comment (kentcdodds) [reply src/index.js:132]: Could we add a comment here indicating why changing the type to text is necessary?

Comment (Meemaw) [reply src/index.js:133]: Done 👍

**Pull request merged**